File: frontend/src/types/projects.ts
export interface ProjectBase {
  title: string;
  description: string;
  year: string;
  url: string;
}

export interface Project extends ProjectBase {
  id: string;
  userId: string;
}

export type ProjectCreate = Partial<ProjectBase>;

export type ProjectUpdate = Partial<ProjectBase>;

export type ProjectsList = Array<Project>;

File: backend/app/schemas/projects.py
from pydantic import BaseModel
from typing import Optional

class ProjectBase(BaseModel):
    title: str
    description: str
    year: str
    url: str

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    year: Optional[str] = None
    url: Optional[str] = None

class ProjectResp(ProjectBase):
    id: str
    user_id: str
    class Config:
        orm_mode = True
